pkg/commands/edit/set/set_name_suffix.go

```diff
@@ -0,0 +1,86 @@
+/*
+Copyright 2018 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package set
+
+import (
+	"errors"
+
+	"github.com/spf13/cobra"
+	"sigs.k8s.io/kustomize/pkg/commands/kustfile"
+	"sigs.k8s.io/kustomize/pkg/fs"
+)
+
+type setNameSuffixOptions struct {
+	suffix string
+}
+
+// newCmdSetNameSuffix sets the value of the nameSuffix field in the kustomization.
+func newCmdSetNameSuffix(fsys fs.FileSystem) *cobra.Command {
+	var o setNameSuffixOptions
+
+	cmd := &cobra.Command{
+		Use:   "namesuffix",
+		Short: "Sets the value of the nameSuffix field in the kustomization file.",
+		Example: `
+The command
+  set namesuffix -acme
+will add the field "nameSuffix: -acme" to the kustomization file if it doesn't exist,
```

Comment:
I'm curious when you run `kustomize edit set namesuffix  -acme`, will it be treated as a flag?

Reply:
Wow, that is correct. Fixed the example!